rmf_rtls/rmf_rtls/database.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2022 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from fastapi import FastAPI
+from tortoise import Tortoise
+from tortoise.contrib.fastapi import register_tortoise
+
+# for tortoise class elements
+from tortoise import fields
+from tortoise.models import Model
+
+from rmf_api_msgs.models import rtls_tag_state, transformation_2D
+
+PydanticRtlsTagState = rtls_tag_state.RtlsTagState
+PydanticTransformation2D = transformation_2D.Transformation2D
+
+###############################################################################
+
+
+class JsonMixin(Model):
+    id = fields.CharField(255, pk=True, source_field="id")
+    data = fields.JSONField()
+
+
+# Note: target_map is unique, thus is set as id - pk
+class TtmTransformation2D(JsonMixin):
+    ref_map = fields.CharField(255, null=True, index=True)
```

Comment:
should these fields to be nullable? ( applies for all other null=True fields following this )

Reply:
Thanks for catching this. Didn't know that it should be non-nullable if in charField, since "" is alr represented as `null`. https://docs.djangoproject.com/en/4.0/ref/models/fields/#django.db.models.Field.null